attest/verifier/types/src/convert/enclave_report_data_contents.rs

```diff
@@ -51,11 +51,32 @@ impl From<&EnclaveReportDataContents> for prost::EnclaveReportDataContents {
         }
     }
 }
+use mc_crypto_digestible::{DigestTranscript, Digestible};
+
+impl Digestible for prost::EnclaveReportDataContents {
+    fn append_to_transcript<DT: DigestTranscript>(
+        &self,
+        context: &'static [u8],
+        transcript: &mut DT,
+    ) {
+        let typename = b"EnclaveReportDataContents";
+        transcript.append_agg_header(context, typename);
+        transcript.append_primitive(context, b"nonce", &self.nonce);
+        transcript.append_primitive(context, b"key", &self.key);
+        // Since custom identity is optional we only include it if it has data.
+        if !self.custom_identity.is_empty() {
+            transcript.append_primitive(context, b"custom_identity", &self.custom_identity);
```

Comment:
There is an `append_to_transcript` impl for `Vec<u8>` but it always uses `bytes` as the typename, which seems like a less than ideal way to ensure consistency. I.e. if prost happens to re-order the members while keeping the tags.